edx_django_utils/data_generation/management/commands/manufacture_data.py

```diff
@@ -0,0 +1,328 @@
+        """
+        return f'<Tree Node {self.data}>'
+
+
+def build_tree_from_field_list(list_of_fields, provided_factory, base_node, customization_value):
+    """
+    Builds a non-binary tree of nodes based on a list of children nodes, using a base node and it's associated data
+    factory as the parent node the user provided value as a reference to a potential, existing record.
+
+    - list_of_fields (list of strings): the linked list of associated objects to create. Example-
+        ['enterprise_customer_user', 'enterprise_customer', 'site']
+    - provided_factory (factory.django.DjangoModelFactory): The data factory of the base_node.
+    - base_node (Node): The parent node of the desired tree to build.
+    - customization_value (string): The value to be assigned to the object associated with the last value in the
+        ``list_of_fields`` param. Can either be a FK if the last value is a subfactory, or alternatively
+        a custom value to be assigned to the field. Example-
+        list_of_fields = ['enterprise_customer_user', 'enterprise_customer', 'site'],
+        customization_value = 9
+        or
+        list_of_fields = ['enterprise_customer_user', 'enterprise_customer', 'name'],
+        customization_value = "FRED"
+    """
+    current_factory = provided_factory
+    current_node = base_node
+    for index, value in enumerate(list_of_fields):
+        try:
+            # First we need to figure out if the current field is a sub factory or not
+            f = getattr(current_factory, value)
+            if isinstance(f, SubFactory):
+                fk_object = None
+                f_model = f.get_factory()._meta.get_model_class()
+
+                # if we're at the end of the list
+                if index == len(list_of_fields) - 1:
+                    # verify that the provided customization value is a valid pk for the model
+                    try:
+                        fk_object = f_model.objects.get(pk=customization_value)
+                    except f_model.DoesNotExist as exc:
+                        raise CommandError(
+                            f"Provided FK value: {customization_value} does not exist on {f_model.__name__}"
+                        ) from exc
+
+                # Look for the node in the tree
+                if node := current_node.find_value(f_model.__name__):
+                    # Not supporting customizations and FK's
+                    if (bool(node.customizations) or bool(node.children)) and bool(fk_object):
+                        raise CommandError("This script does not support customizing provided existing objects")
+                    # If we found the valid FK earlier, assign it to the node
+                    if fk_object:
+                        node.instance = fk_object
+                    # Add the field to the children of the current node
+                    if node not in current_node.children:
+                        current_node.add_child(node)
```

Comment:
same here... how is the node not supposed to be in the children if we're doing `if node := current_node.find_value(f_model.__name__):` above 🤔

Reply:
seems like it's saying "find the node associated with the current model, if that node is not a child add it"... can you find a node that's not a child? Maybe a child's child?

Reply:
aHA! I got it 😄

Reply:
yeah so it's about specifying an object in a chain... I updated the table so that companies and contact infos reference persons then...
```

    def test_command_builds_chains_of_pk(self):
        """
        Test that the manufacture_data command supports customizing nested objects.
        """
        assert TestPerson.objects.all().count() == 0
        assert TestContactInfo.objects.all().count() == 0
        joe = TestPersonFactory()

        created_object = call_command(
            self.command,
            model="edx_django_utils.data_generation.tests.models.TestPersonContactPhoneNumber",
            field_customizations={
                "test_contact_info__test_company__test_person": f"{joe.pk}",
                "test_contact_info__test_person": f"{joe.pk}",
            },
        )
```

Reply:
this actually works for BOTH testing holes above...

Reply:
@alex-sheehan-edx and I went over this offline and determined that these lines aren't reachable at this point, and so we've removed them.